### ISSUE TITLE
Writing profile to disk fails because hashlib does not support usedforsecurity keyword

### ISSUE THREAD

**Issue by Martin-Rey** [pynbody/pynbody#581]
Hi all,

I am trying to write a `pynbody.profile.Profile` object, using its build-in method `write()`:
https://github.com/pynbody/pynbody/blob/6b0ded6b3f21d83b86143c391298bfcb3934b245/pynbody/analysis/profile.py#L481
 which fails with the following trace:
```
Traceback (most recent call last):
  File "/Users/martinrey/Documents/stellar_halos/plot_surface_brightness.py", line 70, in <module>
    profile2d.write()
  File "/opt/anaconda3/lib/python3.7/site-packages/pynbody-0.48-py3.7-macosx-10.9-x86_64.egg/pynbody/analysis/profile.py", line 551, in write
    filename = self._generate_hash_filename()
  File "/opt/anaconda3/lib/python3.7/site-packages/pynbody-0.48-py3.7-macosx-10.9-x86_64.egg/pynbody/analysis/profile.py", line 522, in _generate_hash_filename
    (hashlib.md5(self._x, usedforsecurity=False)).hexdigest()
TypeError: openssl_md5() takes no keyword arguments
```

Thanks to https://stackoverflow.com/questions/54717862/how-do-i-know-if-the-usedforsecurity-flag-is-supported-by-hashlib-md5 and https://github.com/sphinx-doc/sphinx/issues/7611, I have tracked this to the fact that my platform (Mac OS) does not support the `usedforsecurity` keyword used in the hashing here:
https://github.com/pynbody/pynbody/blob/6b0ded6b3f21d83b86143c391298bfcb3934b245/pynbody/analysis/profile.py#L468-L477

From python, `hash lib` is indeed missing the keyword
```
In [1]: import hashlib, inspect                                                                                                        

In [2]: inspect.getfullargspec(hashlib.new)                                                                                            
Out[2]: FullArgSpec(args=['name', 'data'], varargs=None, varkw='kwargs', defaults=(b'',), kwonlyargs=[], kwonlydefaults=None, annotations={})
```

for the following platform configuration
```
In [3]: import sys, platform, pynbody, numpy; print("%s\r\n%s\r\npynbody %s\r\nnumpy %s\r\n"%(sys.version, platform.platform(), pynbody
   ...: .__version__, numpy.__version__))                                                                                              
3.7.4 (default, Aug 13 2019, 15:17:50) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
Darwin-19.6.0-x86_64-i386-64bit
pynbody 0.47
numpy 1.17.2
```

From what I have understood, the keyword `usedforsecurity` is unsafe because not supported across all platforms. But my understanding of hashing algorithms is fairly limited (to say the least) and I don't understand whether it is necessary in this context.

Hope that helps,
Martin

**Comment by apontzen** [comment]
Thanks, are you able to do a pull request and perhaps even a test? This is a feature I've never actually used.

**Comment by Martin-Rey** [comment]
Yes, I can look and figure out a fix to this. As for a test, I will try but this seems to arise from differences in backend implementations on different operating systems, and I am unsure how to capture this in the Travis build. Will have a go anyway.

**Comment by Martin-Rey** [comment]
Solved and merged in #599
